bugz.py

```diff
@@ -0,0 +1,202 @@
+FIELDS = ["id", "summary", "flags", "severity",
+          "priority", "status", "resolution"]
+
+
+class Bugz:
+
+    def __init__(self) -> None:
+        self.conn = BugzillaAPIClient()
+
+    def get_bugs(self, bug_ids: list) -> list:
+        bugs = self.conn.bz_client.getbugs(bug_ids)
+        return bugs
+
+    def build_query(self, query: dict) -> dict:
+        formatted_query = self.conn.bz_client.build_query(query)
+        return formatted_query
+
+    def query(self, query: dict) -> list:
+        bugs = self.conn.bz_client.query(query)
+        return bugs
+
+    def get_query_from_url(self, url: str) -> dict:
+        query = self.conn.bz_client.url_to_query(url)
+        return query
+
+
+class BugzillaHelper:
+    def __init__(self) -> None:
+        self.bugzilla = Bugz()
+
+    def get_bugs(self, bugs: list) -> list:
+        """Get a list of bugs from Bugzilla."""
+        return self.bugzilla.get_bugs(bugs)
+
+    def build_query(self, query: dict) -> dict:
+        """Build a query for Bugzilla."""
+        return self.bugzilla.build_query(query)
+
+    def query(self, query: dict) -> list:
+        """Query Bugzilla."""
+        return self.bugzilla.query(query)
+
+    def get_query_from_url(self, url: str) -> dict:
+        """Get a query from a Bugzilla URL."""
+        return self.bugzilla.get_query_from_url(url)
+
+
+class BugzillaClient(Bugz):
+    def __init__(self):
+        super().__init__()
+        self.db = DatabaseBugzilla()
+        self.BugzillaHelperClient = BugzillaHelper()
+
+    def contains_criteria(self, entry, criteria):
+        return all(entry.get(key) == value for key, value in criteria.items())
+
+    def bugzilla_query(self):
+        allBugs = []
+        for product in PRODUCTS:
+            query = dict(product=product, include_fields=FIELDS)
+            bugs = self.BugzillaHelperClient.query(query)
+
+            for bug in bugs:
+                bug_ = [bug.id, bug.summary, bug.flags,
+                        bug.severity, bug.priority, bug.status, bug.resolution]
+                allBugs.append(bug_)
+
+        return allBugs
+
+    def bugzilla_query_qe_verify(self):
+        qe_bugs = []
+        search_criteria = {'name': 'qe-verify', 'status': '+'}
+
+        payload = self.bugzilla_query()
+
+        for bug in payload:
+            result = any(self.contains_criteria(entry, search_criteria) for entry in bug[2]) # noqa
+            if result:
+                qe_bugs.append(bug)
+
+        return qe_bugs
+
+    def bugzilla_query_severity(self):
+        # payload = self.bugzilla_query()
+
+        # TBD to get all NEW bugs
+        return
+
+    def bugzilla_qe_verify(self):
+        payload = self.bugzilla_query_qe_verify()
+
+        rows = []
+        for entry in payload:
+            bug_id = entry[0]
+            description = entry[1]
+            severity = entry[3]
+            priority = entry[4]
+            status = entry[5]
+            resolution = entry[6]
```

Comment:
I would add some comments here as these indexes could be brittle if non-existent or changed.

Reply:
Same comment as Aaron here.  You can either add comments to things like: entry[2].  or replace those with more meaningful variables like above where you're using bug_id instead of entry[0] ... or both.

Reply:
oh I thought I have added that already.. sorry I missed that, will add comments

Reply:
actually, added an example above for each entry. I get an entry with different fields, which I assing to meaningful variables, entry[0] is going to be the bug_id and so on...